sdk/eventhub/event-hubs/migrationguide.md

```diff
@@ -0,0 +1,210 @@
+
+## Migration samples
+
+* [Receiving events](#migrating-code-from-eventhubclient-to-eventhubconsumerclient-for-receiving-events)
+* [Receiving events with checkpointing](#migrating-code-from-eventprocessorhost-to-eventhubconsumerclient-for-receiving-events)
+* [Sending events](#migrating-code-from-eventhubclient-to-eventhubproducerclient-for-sending-events)
+
+### Migrating code from `EventHubClient` to `EventHubConsumerClient` for receiving events
+
+In V2, event handlers were passed as positional arguments to `receive`.
+
+In V5, event handlers are passed as part of a `SubscriptionEventHandlers` shaped object.
+
+For example, this code which receives from a partition in V2:
+
+```typescript
+const client = EventHubClient.createFromConnectionString(connectionString);
+const rcvHandler = client.receive(partitionId, onMessageHandler, onErrorHandler, {
+  eventPosition: EventPosition.fromStart(),
+  consumerGroup: consumerGroupName
+});
+await rcvHandler.stop();
+```
+
+Becomes this in V5:
+
+```typescript
+const eventHubConsumerClient = new EventHubConsumerClient(consumerGroupName, connectionString);
+
+const subscription = eventHubConsumerClient.subscribe(
+  partitionId, {
+    processInitialize: (initContext) => {
+      initContext.setStartingPosition(EventPosition.fromStart());
+    },
+    processEvents: onMessageHandler,
+    processError: onErrorHandler
+});
+  
+await subscription.close();
+```
+
+See [`receiveEvents.ts`](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) 
+for a sample program demonstrating this.
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for sending events
+
+In V2, there were multiple options on how to send data.
+
+In V5, this has been consolidated into a more efficient `sendBatch` method. 
+Batching merges information from multiple messages into a single send, reducing
+the amount of network communication needed vs sending messages one at a time.
+
+So in V2:
+```typescript
+const eventsToSend = [
+  // events go here
+];
+
+const client = EventHubClient.createFromConnectionString(connectionString);
+
+// Would fail if the total size of events exceed the max size supported by the library.
+await client.sendBatch(eventsToSend, partitionId);
+```
+
+In V5:
+```typescript
+const producer = new EventHubProducerClient(connectionString);
+
+const eventsToSend = [
+  // events go here
+];
+
+let batch = await producer.createBatch();
+let i = 0;
+
+while (i < eventsToSend.length) {
+  // messages can fail to be added to the batch if they exceed the maximum size configured for
+  // the EventHub.
+  const isAdded = batch.tryAdd(eventsToSend[i]);
+
+  if (isAdded) {
+    console.log(`Added eventsToSend[${i}]`);
```

Comment:
```suggestion
    console.log(`Added event number ${i} to the batch`);
```

It would be good to match this up with a `console.log(`Creating a new batch`)` for everytime we create a new batch.
Then the logs would show a nice story of batches being created, filled and sent